src/TwigComponent/src/ComponentFactory.php

```diff
@@ -217,6 +217,32 @@ private function isAnonymousComponent(string $name): bool
      */
     private function throwUnknownComponentException(string $name): void
     {
-        throw new \InvalidArgumentException(sprintf('Unknown component "%s". The registered components are: %s. And no matching anonymous component template was found', $name, implode(', ', array_keys($this->config))));
+        $message = sprintf('Unknown component "%s".', $name);
+        $lowerName = strtolower($name);
+        $nameLength = \strlen($lowerName);
+        $alternatives = [];
+
+        foreach (array_keys($this->config) as $type) {
+            $lowerType = strtolower($type);
+            $lev = levenshtein($lowerName, $lowerType);
+
+            if ($lev <= $nameLength / 3 || str_contains($lowerType, $lowerName)) {
+                $alternatives[] = $type;
+            }
```

Comment:
What about low cost check first instead of doing both?
```suggestion
            if (str_contains($lowerType, $lowerName)) {
                $alternatives[] = $type;
            } elseif (levenshtein($lowerName, $lowerType) <= $nameLength / 3) {
                $alternatives[] = $type;
            }
```

Reply:
Are you going to get this exception all the time? :slightly_smiling_face: